### PR TITLE
hack,test: initial e2e test for helm operator

### PR DESCRIPTION
**Pull request by joelanford**
This PR:
* Adds new scripts to: 
  * setup OpenShift
  * build the helm-app-operator binary and base image
  * run unit tests
  * build a memcached-operator image from the base image and the official memcached helm chart
  * deploy a memcached helm release via a custom resource
  * verify that the helm release resources are deployed
* Updates .travis.yml to run the above scripts for CI

### CODE REVIEW

File: helm-app-operator/hack/tests/sanity-check.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+set -ex
+
+go vet ./...
+
+# TODO(joelanford): add license headers to files
```
Comment (joelanford) [inline helm-app-operator/hack/tests/sanity-check.sh]: Need to add license headers to the helm-app-operator files.

File: helm-app-operator/hack/tests/sanity-check.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#./hack/check_license.sh
+
+# TODO(joelanford): fix case of error messages
```
Comment (joelanford) [inline helm-app-operator/hack/tests/sanity-check.sh]: Need to update log and error message casing to conform to SDK standard.

File: helm-app-operator/hack/tests/sanity-check.sh (hunk 1 above)
Comment (joelanford) [reply helm-app-operator/hack/tests/sanity-check.sh]: See #54

File: helm-app-operator/hack/tests/sanity-check.sh (hunk 2 above)
Comment (joelanford) [reply helm-app-operator/hack/tests/sanity-check.sh]: See #55

**Comment by joelanford** [comment]
I made a bunch of updates to make this repo look as similar to the SDK repo as possible, including proposed changes from operator-framework/operator-sdk#631. I think it's mostly ready, but I'll leave in WIP until that PR is accepted.

**Comment by joelanford** [comment]
The latest commit has a fix for the race condition that was causing flakes in the e2e. Should be ready for review again.

**Comment by joelanford** [comment]
Uncommented and fixed sanity check tests for license headers and error casing to fix #54 and #55

File: helm-app-operator/Makefile [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+test/unit:
+	./hack/tests/unit.sh
+
+test/e2e: test/e2e/helm
+
+test/e2e/helm:
+	./hack/tests/e2e-helm.sh
+
+.PHONY: test test/sanity test/unit test/e2e test/e2e/helm test/ci-helm
```
Comment (hasbro17) [inline helm-app-operator/Makefile]: Leave a newline at the end.

File: helm-app-operator/hack/ci/setup-openshift.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Configure insecure docker registry for openshift
+sudo service docker stop
+sudo sed -i 's/DOCKER_OPTS=\"/DOCKER_OPTS=\"--insecure-registry 172.30.0.0\/16 /' /etc/default/docker
+sudo service docker start
+# Download oc to spin up openshift on local docker instance
+curl -Lo oc.tar.gz https://github.com/openshift/origin/releases/download/v3.10.0/openshift-origin-client-tools-v3.10.0-dd10d17-linux-64bit.tar.gz
```
Comment (hasbro17) [inline helm-app-operator/hack/ci/setup-openshift.sh]: We can use `v3.11.0`
https://github.com/openshift/origin/releases/tag/v3.11.0

I think that's what we're testing the SDK against.

File: helm-app-operator/hack/tests/e2e-helm.sh [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+# wait for operator pod to run
+if ! timeout 1m kubectl rollout status deployment/memcached-operator;
+then
+    kubectl describe deployment memcached-operator
+    kubectl logs deployment/memcached-operator
+    exit 1
+fi
+
+# create CR
+kubectl create -f deploy/cr.yaml
```
Comment (hasbro17) [inline helm-app-operator/hack/tests/e2e-helm.sh]: Btw this is fine for now but just a heads up when we move this test into the SDK the project layout is a little different after the controller-runtime refactor:
https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator/deploy

**Pull request merged**